webplatformcompat/renderers.py

```diff
@@ -1,244 +1,242 @@
-        response = renderer_context.get("response", None)
-        status_code = response and response.status_code
-        if status_code != 400:
-            raise WrapperNotApplicable('Status code must be 400.')
-        if not (data and '_view_extra' in data):
-            raise WrapperNotApplicable('Not linked results')
-
-        view_extra = data.pop('_view_extra')
-        assert isinstance(view_extra, dict)
-        converted = {}
-        for rname, error_dict in view_extra.items():
-            assert rname != 'meta'
-            for seq, errors in error_dict.items():
-                if seq is None:  # pragma: no cover
-                    # TODO: diagnose how subject feature errors are getting
-                    # into view_extra.
-                    seq = "subject"
-                for fieldname, error in errors.items():
-                    name = 'linked.%s.%s.%s' % (rname, seq, fieldname)
-                    converted[name] = error
-        data.update(converted)
-
-        return self.wrap_error(
-            data, renderer_context, keys_are_fields=True, issue_is_title=False)
-
-    def model_to_resource_type(self, model):
-        if model:
-            return snakecase(model._meta.verbose_name_plural)
-        else:
-            return 'data'
+    media_type = 'application/vnd.api+json'
 
-    def handle_related_field(self, resource, field, field_name, request):
-        """Handle PrimaryKeyRelatedField
+    def render(self, data, accepted_media_type=None, renderer_context=None):
+        """Convert native DRF data to JSON API RC1.
 
-        Same as base handle_related_field, but:
-        - adds href to links, using DRF default name
-        - doesn't handle data not in fields
-        - uses presence of child_relation attribute to signify "many"
+        The four expected kinds of data are:
+        - No data (maybe in response to 204 No Content)
+        - Error data (status code is 4xx or 5xx)
+        - Standard data (with serializer and fields_extra)
+        - Paginated data
         """
-        links = self.dict_class()
-        linked_ids = self.dict_class()
-
-        many = hasattr(field, 'child_relation')
-        model = self.model_from_obj(field)
-        if model:
-            resource_type = self.model_to_resource_type(model)
-
-            format_kwargs = {
-                'model_name': model._meta.object_name.lower()
-            }
-            view_name = '%(model_name)s-detail' % format_kwargs
-
-            links[field_name] = self.dict_class((
-                ("type", resource_type),
-                ("href", self.url_to_template(view_name, request, field_name)),
-            ))
-
-        assert field_name in resource
-        if many:
-            pks = resource[field_name]
+        response = renderer_context.get('response')
+        request = renderer_context.get('request')
+        fields_extra = renderer_context.get('fields_extra')
+        status_code = response and response.status_code
```

Comment:
Good question. I don't know if it is valid to call the parser without a response in the context. I can't trip that state with integration tests or a few manual tests. Maybe the line should change to

```
response = renderer_context['response']
```
and we'll see if the universe can tell us when it happens.